### PR TITLE
FEATURE: Add settings to customize login screen stylesheets

### PR DESCRIPTION
**Pull request by dfeyer**

### CODE REVIEW

**Comment by dfeyer** [comment]
Based on https://discuss.neos.io/t/how-to-change-the-login-page-background/523/3, but a lots more easy to configure now.

I target it to 2.1, can be nice to have a custom login screen for this release, like @robertlemke suggest on Discuss. I know I'm late for the feature freeze, so completly fine to retarget this PR on master.

From my side :+1:

File: TYPO3.Neos/Configuration/Settings.yaml [hunk 1]
```diff
@@ -397,6 +397,10 @@ TYPO3:
             authenticationProviderName: '${entity.authenticationProviderName}'
             name: '${entity.party.name.fullName}'
 
+    login:
+      styles:
```
Comment (bwaidelich) [inline TYPO3.Neos/Configuration/Settings.yaml]: I don't have a better suggestion, but I find `TYPO3.Neos.login.styles` a bit confusing for this setting:
"login" is very generic, but it only targets the *Neos backend login form*, right? So maybe call it `backendLoginForm`?
And "stylesheets" would be more in sync with [Neos templating] (http://neos.readthedocs.org/en/latest/CreatingASite/RenderingCustomMarkup/AdjustingOutput.html#the-head) than "styles"

Comment (dimaip) [inline TYPO3.Neos/Configuration/Settings.yaml]: Agree

File: TYPO3.Neos/Configuration/Settings.yaml [hunk 2]
```diff
@@ -397,6 +397,10 @@ TYPO3:
...
+        'resource://TYPO3.Neos/Public/Styles/Login.css': TRUE
```
Comment (bwaidelich) [inline TYPO3.Neos/Configuration/Settings.yaml]: This creates a coupling to the file path/name. It's probably not really an issue in practice, but if the path changes for some reason, a different setting that overwrites it would be appended instead of replaced.
Again, more a theoretical problem, but something like this could work as well:
```
'TYPO3.Neos:DefaultStyles': 'resource://TYPO3.Neos/Public/Styles/Login.css'
```

**Comment by bwaidelich** [comment]
Nice feature, thanks a lot for taking care!
Two minor issues I have with the settings style, but nothing that should block this from being merged!

**Comment by dfeyer** [comment]
@bwaidelich @dimaip Fixed based on your comments, thanks

**Comment by dimaip** [comment]
Thanks a lot for this feature, very nice!
Frankly I don't see any formal justification for merging it after feature-freeze, unless we ship custom login-screen image with every release, which I hope we don't! :)
I would target it for master, but not hard point here.

**Comment by bwaidelich** [comment]
@dfeyer wow, you're faster than light!
But you missed my second comment which was referring to the settings, too:

```
This creates a coupling to the file path/name. It's probably not really an issue in practice, but if the path changes for some reason, a different setting that overwrites it would be appended instead of replaced.
Again, more a theoretical problem, but something like this could work as well:

'TYPO3.Neos:DefaultStyles': 'resource://TYPO3.Neos/Public/Styles/Login.css'
```

BTW: Do you always force-push to squash your commits?

**Comment by bwaidelich** [comment]
and BTW2: I agree with @dimaip here, it should be for master

**Comment by dimaip** [comment]
Ok, one more BTW: this PR is missing a description :)

**Comment by dfeyer** [comment]
@bwaidelich I use force push for small change only now. Too hard to follow on bigger change. But when the force push mark some "diff comment" as outdated, it's OK. 

And OK I close this one a push a new one to master ;)
